public/components/common/modals/model_version_error_details_modal.tsx

```diff
@@ -22,37 +22,52 @@ import {
 
 import { routerPaths } from '../../../../common/router_paths';
 
+const mode2ErrorTitleMap = {
+  'deployment-failed': 'deployment failed',
+  'artifact-upload-failed': 'artifact upload failed',
+  'undeployment-failed': 'undeployment failed',
+};
+
 export const ModelVersionErrorDetailsModal = ({
   id,
+  mode,
   name,
   version,
   closeModal,
   errorDetails,
-  isDeployFailed,
+  plainVersionLink,
 }: {
-  name: string;
   id: string;
+  mode: 'deployment-failed' | 'artifact-upload-failed' | 'undeployment-failed';
+  name: string;
   version: string;
   closeModal: () => void;
   errorDetails: string;
-  isDeployFailed?: boolean;
+  plainVersionLink?: string;
```

Comment:
The `plainVersionLink` isn't the same as `generatePath(routerPaths.modelVersion, { id })`. The `plainVersionLink` was created by `history.createHref` and will include the basename part. We can't use this link in the react-router `Link` component. We can't use react-router component in `overlays?.openModal`. What about remove the Link component in the error details modal?

Reply:
Thanks for the explanation!

> What about remove the Link component in the error details modal?
You mean we  always use `plainVersionLink`?

Reply:
Yes. But the page will be blank and re-enter after click the link element.

Reply:
@wanglam If we mount the Modal in the component where it's used(NOT via overlays), I suppose then we can use react-router Link, will that solve the reload problem?

Reply:
Yes. It will a little bit complicated to render this component in the render tree. Maybe we need to create a model error details context and render this component based the context in the model page level.

Reply:
Sounds good to me, I'm fine with the current implementation, we can look for improvements in the future